sql/internal/sqlx/diff.go

```diff
@@ -476,6 +476,28 @@ func Has(elements, target any) bool {
 	return false
 }
 
+// HasCharset reports if the attribute contains the "charset" attribute,
+// and it needs to be defined explicitly on the schema. This is true, in
+// case the element charset is different from its parent charset.
+func HasCharset(attr []schema.Attr, parent []schema.Attr) (string, bool) {
+	var c, p schema.Charset
+	if Has(attr, &c) && (parent == nil || Has(parent, &p) && c.V != p.V) {
+		return c.V, true
+	}
+	return "", false
+}
+
+// HasCollate reports if the attribute contains the "collation"/"collate" attribute,
+// and it needs to be defined explicitly on the schema. This is true, in
+// case the element collation is different from its parent collation.
+func HasCollate(attr []schema.Attr, parent []schema.Attr) (string, bool) {
```

Comment:
If it's copied from mysql, let's do it in a separate PR and reuse it there as well.

Reply:
Sure, will move to another pr for small work on this